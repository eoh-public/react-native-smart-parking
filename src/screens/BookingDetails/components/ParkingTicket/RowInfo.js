import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';
import InfoField from 'components/SmartParking/BookingDetails/InfoField';

const RowInfo = memo(
  ({ leftValue, rightValue, leftTitle, rightTitle, body = true }) => {
    return (
      <View style={styles.container}>
        <InfoField title={leftTitle} value={leftValue} body={body} />
        <InfoField
          title={rightTitle}
          value={rightValue}
          style={styles.right}
          body={body}
        />
      </View>
    );
  }
);

export default RowInfo;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  text: {
    marginTop: 8,
  },
  right: {
    paddingLeft: 16,
  },
});
