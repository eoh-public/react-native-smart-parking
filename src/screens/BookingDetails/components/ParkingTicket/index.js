import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';
import {t} from 'i18n-js';;

import { Colors } from '../../../../configs';
import Text from '../../../../commons/Text';
import TicketViewSpace from '../TicketViewSpace';
import TimeCountDown from '../TimeCountDown';
import RowInfo from './RowInfo';
import InfoField from '../InfoField';
import { openMapDirection, formatMoney } from '../../../../utils/Utils';

const ParkingTicket = memo(
  ({
    time_remaining,
    start_countdown,
    arrive_at,
    leave_at,
    pay_before,
    plate_number,
    spot_name,
    parking_area,
    parking_address,
    parking_lat,
    parking_lng,
    status,
    is_paid,
    grand_total,
    getBookingDetail,
  }) => {
    const timeFormat = 'LT - DD/MM/YYYY';
    const arrive_at_str = arrive_at && arrive_at.format(timeFormat);
    const leave_at_str = leave_at && leave_at.format(timeFormat);
    const pay_before_str = pay_before && pay_before.format(timeFormat);

    return (
      <View style={styles.container}>
        <View>
          <View style={styles.ticketTopContainer}>
            <Text type="Body" semibold>
              {t('parking_session')}
            </Text>
            {status === '' && (
              <TimeCountDown
                time_remaining={time_remaining}
                start_countdown={start_countdown}
                getBookingDetail={getBookingDetail}
              />
            )}
            <RowInfo
              leftValue={arrive_at_str}
              rightValue={leave_at_str}
              leftTitle={t('arrive_at')}
              rightTitle={t('leave_at')}
            />
          </View>
          <TicketViewSpace />
          <View style={styles.ticketBottomContainer}>
            <Text type={'Body'} semibold>
              {t('parking_information')}
            </Text>
            <RowInfo
              leftValue={plate_number}
              rightValue={spot_name}
              rightTitle={t('parking_spot')}
              leftTitle={t('license_plate')}
              body={false}
            />
            <InfoField
              title={t('parking_area')}
              value={parking_area}
              style={styles.infoField}
            />
            <InfoField
              title={t('parking_address')}
              value={parking_address}
              style={styles.infoField}
              onDirection={openMapDirection({
                lat: parking_lat,
                lng: parking_lng,
              })}
            />
            <View style={styles.totalPay}>
              <View style={styles.totalRow}>
                <Text type="Body" color={Colors.Gray8}>
                  {t('text_total')}
                </Text>
                <Text type="H4" color={Colors.Orange} semibold>
                  {formatMoney(grand_total)}
                </Text>
              </View>
              {!is_paid && status === '' && (
                <Text
                  color={Colors.Red6}
                  type="Body"
                  style={styles.textPayBefore}
                >
                  {t('pay_before_%{time}', { time: pay_before_str })}
                </Text>
              )}
            </View>
          </View>
        </View>
      </View>
    );
  }
);

export default ParkingTicket;

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.White,
    marginVertical: 16,
    marginHorizontal: 16,
    borderRadius: 8,
  },
  ticketTopContainer: {
    paddingTop: 16,
    paddingBottom: 8,
    paddingHorizontal: 16,
    borderTopLeftRadius: 8,
    borderTopRightRadius: 8,
    borderWidth: 1,
    borderBottomWidth: 0,
    borderColor: Colors.Gray4,
  },
  ticketBottomContainer: {
    paddingBottom: 24,
    paddingHorizontal: 16,
    borderBottomLeftRadius: 8,
    borderBottomRightRadius: 8,
    borderWidth: 1,
    borderTopWidth: 0,
    borderColor: Colors.Gray4,
  },
  infoField: {
    marginTop: 4,
  },
  totalPay: {
    marginTop: 17,
    paddingTop: 17,
    paddingBottom: 8,
    borderTopWidth: 1,
    borderColor: Colors.Gray4,
  },
  totalRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  textPayBefore: {
    textAlign: 'center',
  },
});
