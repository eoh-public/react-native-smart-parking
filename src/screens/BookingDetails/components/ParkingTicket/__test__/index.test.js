import React from 'react';
import renderer, { act } from 'react-test-renderer';
import ParkingTicket from '..';
import moment from 'moment';

describe('Test ParkingTicket', () => {
  let data;

  beforeEach(() => {
    Date.now = jest.fn(() => new Date('2021-01-24T12:00:00.000Z'));
    data = {
      time_remaining: 100,
      start_countdown: false,
      arrive_at: moment(),
      leave_at: moment(),
      pay_before: moment(),
      plate_number: '59Z-1234',
      spot_name: 'HU1',
      parking_area: '',
      parking_address: '',
      parking_lat: '123',
      parking_lng: '456',
      status: '',
      is_paid: false,
      grand_total: 1,
    };
  });

  let tree;

  test('render', () => {
    act(() => {
      tree = renderer.create(<ParkingTicket {...data} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
