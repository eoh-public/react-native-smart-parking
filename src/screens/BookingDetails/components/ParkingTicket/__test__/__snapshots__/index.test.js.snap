// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ParkingTicket render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "marginHorizontal": 16,
      "marginVertical": 16,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#E8E8E8",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderWidth": 1,
          "paddingBottom": 8,
          "paddingHorizontal": 16,
          "paddingTop": 16,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Semibold",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Thời gian đỗ xe
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#FAFAFA",
            "borderColor": "#E8E8E8",
            "borderRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 10,
            "paddingVertical": 24,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 2,
                  "elevation": 1,
                  "height": 48,
                  "marginRight": 4,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 2,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 2,
                  "width": 32,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F5F5F5",
                    "borderTopLeftRadius": 2,
                    "borderTopRightRadius": 2,
                    "height": 24,
                    "position": "absolute",
                    "width": 32,
                  }
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "#E8E8E8",
                    "borderBottomLeftRadius": 2,
                    "borderBottomRightRadius": 2,
                    "height": 24,
                    "position": "absolute",
                    "top": 24,
                    "width": 32,
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#F5222D",
                      "fontFamily": "SFProDisplay-Semibold",
                      "fontSize": 24,
                      "lineHeight": 32,
                      "textAlign": "left",
                      "textDecorationLine": "none",
                    },
                    Object {
                      "marginTop": 8,
                      "textAlign": "center",
                      "width": 32,
                    },
                  ]
                }
                testID="TIME_COUNT_DOWN_TEXT"
              >
                0
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 2,
                  "elevation": 1,
                  "height": 48,
                  "marginRight": 4,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 2,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 2,
                  "width": 32,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F5F5F5",
                    "borderTopLeftRadius": 2,
                    "borderTopRightRadius": 2,
                    "height": 24,
                    "position": "absolute",
                    "width": 32,
                  }
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "#E8E8E8",
                    "borderBottomLeftRadius": 2,
                    "borderBottomRightRadius": 2,
                    "height": 24,
                    "position": "absolute",
                    "top": 24,
                    "width": 32,
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#F5222D",
                      "fontFamily": "SFProDisplay-Semibold",
                      "fontSize": 24,
                      "lineHeight": 32,
                      "textAlign": "left",
                      "textDecorationLine": "none",
                    },
                    Object {
                      "marginTop": 8,
                      "textAlign": "center",
                      "width": 32,
                    },
                  ]
                }
                testID="TIME_COUNT_DOWN_TEXT"
              >
                0
              </Text>
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#595959",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 14,
                  "lineHeight": 22,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            giờ
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginLeft": 4,
                "marginRight": 8,
                "marginTop": 8,
              },
            ]
          }
        >
          :
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 2,
                  "elevation": 1,
                  "height": 48,
                  "marginRight": 4,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 2,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 2,
                  "width": 32,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F5F5F5",
                    "borderTopLeftRadius": 2,
                    "borderTopRightRadius": 2,
                    "height": 24,
                    "position": "absolute",
                    "width": 32,
                  }
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "#E8E8E8",
                    "borderBottomLeftRadius": 2,
                    "borderBottomRightRadius": 2,
                    "height": 24,
                    "position": "absolute",
                    "top": 24,
                    "width": 32,
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#F5222D",
                      "fontFamily": "SFProDisplay-Semibold",
                      "fontSize": 24,
                      "lineHeight": 32,
                      "textAlign": "left",
                      "textDecorationLine": "none",
                    },
                    Object {
                      "marginTop": 8,
                      "textAlign": "center",
                      "width": 32,
                    },
                  ]
                }
                testID="TIME_COUNT_DOWN_TEXT"
              >
                0
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 2,
                  "elevation": 1,
                  "height": 48,
                  "marginRight": 4,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 2,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 2,
                  "width": 32,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F5F5F5",
                    "borderTopLeftRadius": 2,
                    "borderTopRightRadius": 2,
                    "height": 24,
                    "position": "absolute",
                    "width": 32,
                  }
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "#E8E8E8",
                    "borderBottomLeftRadius": 2,
                    "borderBottomRightRadius": 2,
                    "height": 24,
                    "position": "absolute",
                    "top": 24,
                    "width": 32,
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#F5222D",
                      "fontFamily": "SFProDisplay-Semibold",
                      "fontSize": 24,
                      "lineHeight": 32,
                      "textAlign": "left",
                      "textDecorationLine": "none",
                    },
                    Object {
                      "marginTop": 8,
                      "textAlign": "center",
                      "width": 32,
                    },
                  ]
                }
                testID="TIME_COUNT_DOWN_TEXT"
              >
                1
              </Text>
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#595959",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 14,
                  "lineHeight": 22,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            phút
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginLeft": 4,
                "marginRight": 8,
                "marginTop": 8,
              },
            ]
          }
        >
          :
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 2,
                  "elevation": 1,
                  "height": 48,
                  "marginRight": 4,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 2,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 2,
                  "width": 32,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F5F5F5",
                    "borderTopLeftRadius": 2,
                    "borderTopRightRadius": 2,
                    "height": 24,
                    "position": "absolute",
                    "width": 32,
                  }
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "#E8E8E8",
                    "borderBottomLeftRadius": 2,
                    "borderBottomRightRadius": 2,
                    "height": 24,
                    "position": "absolute",
                    "top": 24,
                    "width": 32,
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#F5222D",
                      "fontFamily": "SFProDisplay-Semibold",
                      "fontSize": 24,
                      "lineHeight": 32,
                      "textAlign": "left",
                      "textDecorationLine": "none",
                    },
                    Object {
                      "marginTop": 8,
                      "textAlign": "center",
                      "width": 32,
                    },
                  ]
                }
                testID="TIME_COUNT_DOWN_TEXT"
              >
                4
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 2,
                  "elevation": 1,
                  "height": 48,
                  "marginRight": 4,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 2,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 2,
                  "width": 32,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F5F5F5",
                    "borderTopLeftRadius": 2,
                    "borderTopRightRadius": 2,
                    "height": 24,
                    "position": "absolute",
                    "width": 32,
                  }
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "#E8E8E8",
                    "borderBottomLeftRadius": 2,
                    "borderBottomRightRadius": 2,
                    "height": 24,
                    "position": "absolute",
                    "top": 24,
                    "width": 32,
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#F5222D",
                      "fontFamily": "SFProDisplay-Semibold",
                      "fontSize": 24,
                      "lineHeight": 32,
                      "textAlign": "left",
                      "textDecorationLine": "none",
                    },
                    Object {
                      "marginTop": 8,
                      "textAlign": "center",
                      "width": 32,
                    },
                  ]
                }
                testID="TIME_COUNT_DOWN_TEXT"
              >
                0
              </Text>
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#595959",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 14,
                  "lineHeight": 22,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            giây
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#8C8C8C",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 14,
                  "lineHeight": 22,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {
                  "marginTop": 8,
                },
              ]
            }
          >
            Bắt đầu
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#262626",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 14,
                  "lineHeight": 22,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {
                  "marginTop": 8,
                },
              ]
            }
          >
            12:00 PM - 24/01/2021
                
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "paddingLeft": 16,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#8C8C8C",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 14,
                  "lineHeight": 22,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {
                  "marginTop": 8,
                },
              ]
            }
          >
            Kết thúc
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#262626",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 14,
                  "lineHeight": 22,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {
                  "marginTop": 8,
                },
              ]
            }
          >
            12:00 PM - 24/01/2021
                
          </Text>
        </View>
      </View>
    </View>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={32}
      bbWidth={718}
      fill="none"
      focusable={false}
      height={32}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 32,
            "width": 718,
          },
        ]
      }
      vbHeight={32}
      vbWidth={718}
      width={718}
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGCircle
          cx="0"
          cy="16"
          propList={
            Array [
              "stroke",
            ]
          }
          r="16"
          stroke={4293454056}
        />
        <RNSVGCircle
          cx="718"
          cy="16"
          propList={
            Array [
              "stroke",
            ]
          }
          r="16"
          stroke={4293454056}
        />
        <RNSVGPath
          d="M16 16 H702"
          propList={
            Array [
              "stroke",
              "strokeDasharray",
            ]
          }
          stroke={4293454056}
          strokeDasharray={
            Array [
              "10",
              "10",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        Object {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderColor": "#E8E8E8",
          "borderTopWidth": 0,
          "borderWidth": 1,
          "paddingBottom": 24,
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Semibold",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Thông tin đỗ xe
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#8C8C8C",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 14,
                  "lineHeight": 22,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {
                  "marginTop": 8,
                },
              ]
            }
          >
            Biển số xe
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#262626",
                  "fontFamily": "SFProDisplay-Semibold",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {
                  "marginTop": 8,
                },
              ]
            }
          >
            59Z-1234
                
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "paddingLeft": 16,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#8C8C8C",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 14,
                  "lineHeight": 22,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {
                  "marginTop": 8,
                },
              ]
            }
          >
            Vị trí đỗ
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#262626",
                  "fontFamily": "SFProDisplay-Semibold",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {
                  "marginTop": 8,
                },
              ]
            }
          >
            HU1
                
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
              },
            ]
          }
        >
          Địa điểm đỗ
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#262626",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
              },
            ]
          }
        >
          
              
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
              },
            ]
          }
        >
          Địa chỉ
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#262626",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
              },
            ]
          }
        >
          
              
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#00979D",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 14,
                  "lineHeight": 22,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            Chỉ đường
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "borderColor": "#E8E8E8",
            "borderTopWidth": 1,
            "marginTop": 17,
            "paddingBottom": 8,
            "paddingTop": 17,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#595959",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 14,
                  "lineHeight": 22,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            Tổng cộng
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#F08229",
                  "fontFamily": "SFProDisplay-Semibold",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {},
              ]
            }
          >
            1 đ
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#F5222D",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "textAlign": "center",
              },
            ]
          }
        >
          Thanh toán trước 12:00 PM - 24/01/2021
        </Text>
      </View>
    </View>
  </View>
</View>
`;
